File: AkitaModel/VoucherPayment.cs
namespace Skyware.Lis.AkitaModel;

/// <summary>
/// Represents a payment made with a voucher.
/// </summary>
public class VoucherPayment : PaymentBase
{

    /// <inheritdoc />
    public override PaymentMethod Method => PaymentMethod.Voucher;

    /// <summary>
    /// Voucher serial number used for the payment.
    /// </summary>
    public string VoucherNumber { get; set; }

}

File: AkitaMotelTS/src/VoucherPayment.ts
/**
 * Skyware.Lis.AkitaModel
 */

import { PaymentBase } from './PaymentBase';
import { PaymentMethod } from './PaymentMethod';

/**
 * Represents a payment made with a voucher.
 */
export class VoucherPayment extends PaymentBase {
  /** @inheritdoc */
  get method(): PaymentMethod {
    return PaymentMethod.Voucher;
  }

  /**
   * Voucher serial number used for the payment.
   */
  voucherNumber?: string;
}
